rafiki/admin/admin.py

```diff
@@ -282,7 +319,25 @@ def create_inference_job(self, user_id, app, app_version):
         if train_job is None:
             raise InvalidTrainJobError('Have you started a train job for this app?')
 
-        if train_job.status != TrainJobStatus.COMPLETED:
+        count = {
+            TrainJobStatus.STARTED: 0,
+            TrainJobStatus.RUNNING: 0,
+            TrainJobStatus.STOPPED: 0,
+            TrainJobStatus.COMPLETED: 0
+        }
+        sub_train_jobs = self._db.get_sub_train_jobs_of_train_job(train_job.id)
+        for sub_train_job in sub_train_jobs:
+            count[sub_train_job.status] += 1
+
+        status = TrainJobStatus.STARTED
+        if count[TrainJobStatus.RUNNING] > 0:
+            status = TrainJobStatus.RUNNING
+        elif count[TrainJobStatus.COMPLETED] == len(sub_train_jobs):
+            status = TrainJobStatus.COMPLETED
+        elif count[TrainJobStatus.STOPPED] > 0:
+            status = TrainJobStatus.STOPPED
+
+        if status != TrainJobStatus.COMPLETED:
```

Comment:
Repeated logic for determining train job status - abstract it out as a method?

Reply:
Extracted as a method.